R/shiny_fuse.R

```diff
@@ -0,0 +1,651 @@
+  shinyfuse_ui <- shinydashboard::dashboardPage(
+    skin = "black",
+
+    # header definition -------------------------------------------------------
+    header = shinydashboard::dashboardHeader(
+      title = "shiny_fuse",
+      titleWidth = 350,
+      shinydashboard::dropdownMenu(
+        type = "tasks",
+        icon = icon("question-circle fa-1g"),
+        badgeStatus = NULL,
+        headerText = "Documentation",
+        shinydashboard::notificationItem(
+          text = actionButton(
+            "interface_overview", "Overview of the interface",
+            icon("hand-o-right")
+          ),
+          icon = icon(""), # tricking it to not have additional icon
+          status = "primary"
+        )
+      )
+    ),
+
+    # sidebar definition ------------------------------------------------------
+    sidebar = shinydashboard::dashboardSidebar(
+      width = 250,
+      collapsed = !is.null(out_annofuse),
+      
+      uiOutput("choose_annofusedata_file"),
+      # shinydashboard::menuItem(
+      #   text = "Input Settings", icon = icon("cog"),
+      #   startExpanded = TRUE,
+      #   numericInput(
+      #     inputId = "whatevs",
+      #     label = "number of genesets",
+      #     value = 15, min = 1, max = 50
+      #   )
+      # )
+      uiOutput("plot_controls")
+    ),
+
+    # body definition ---------------------------------------------------------
+    body = shinydashboard::dashboardBody(
+      rintrojs::introjsUI(),
+      ## Define output size and style of error messages
+      shiny::tags$head(
+        shiny::tags$style(
+          shiny::HTML(".shiny-output-error-validation {
+                 font-size: 15px;
+                 color: forestgreen;
+                 text-align: center;
+                 }
+                 ")
+        )
+      ),
+      id = "main-app",
+      navbarPage(
+        title = div(),
+        windowTitle = "shinyfuse",
+        footer = "",
+        theme = shinytheme("cosmo"),
+        selected = "TableExplorer",
+
+        # ui TableExplorer -----------------------------------------------------
+        tabPanel(
+          title = "TableExplorer", icon = icon("table"),
+          fluidPage(
+            fluidRow(
+              column(
+                width = 8,
+                DT::dataTableOutput("table_annofuse")
+              ),
+              column(
+                width = 4,
+                fluidRow(
+                  column(
+                    width = 4,
+                    offset = 4,
+                    actionButton("btn_load_exonsdata", "load exons")
+                  ),
+                  column(
+                    width = 4,
+                    actionButton("btn_load_pfamdata", "load pfam")
+                  )
+                ),
+                h4("Some content, for example linked to the selected row"),
+                uiOutput("geneinfo_ui"),
+                h4("Expanding more on this..."),
+                uiOutput("geneplots_ui")
```

Comment:
can we give width param here for the plot? Sometime when the domain names are too long there seems to be no space for the actual plot like:
<img width="385" alt="Screen Shot 2020-08-14 at 11 38 49 AM" src="https://user-images.githubusercontent.com/34580719/90268319-d875a900-de24-11ea-8bdc-b3e22d5635bc.png">

Or I can add a quick fix of adding the legend to the bottom in the plot_breakpoints() function. Let me know what will work best. Thanks!

Reply:
The plots are quite squished due to the narrow shiny app window.
Probably the option to put it at the bottom is best, given that it is often with long legend names.
It can be IMHO at best as a parameter that tweaks the placement